src/main/features/response/helpers/taskListBuilder.ts

```diff
@@ -14,25 +14,20 @@ import { WhenWillYouPayTask } from 'response/tasks/whenWillYouPayTask'
 import { FreeMediationTask } from 'response/tasks/freeMediationTask'
 import { RejectPartOfClaimOption } from 'response/form/models/rejectPartOfClaim'
 import { Claim } from 'claims/models/claim'
+import { WhenDidYouPayTask } from 'response/tasks/whenDidYouPayTask'
 
 export class TaskListBuilder {
-  static buildBeforeYouStartSection (draft: ResponseDraft, externalId: string): TaskList {
+  static buildBeforeYouStartSection (draft: ResponseDraft, claim: Claim): TaskList {
```

Comment:
why is this changed?
it looks very strange down on line 135 where one of these methods takes an external id and one takes
REQ

Reply:
@timja-kainos I asked him to pass only what is needed.

That method used to need external ID only to render the links correctly and `buildRespondToClaimSection` used to display 'More time needed to respond' task where response deadline retrieved from claim instance was used to make display / no display decision. Now since 'More time needed to respond' was moved up to 'Before you start' section the `buildBeforeYouStartSection` method needs claim instance.

On the second thought to stick to 'pass as little as possible rule' we could have the following signature:

```
static buildBeforeYouStartSection (draft: ResponseDraft, externalId: string, responseDeadline: Moment): TaskList
```

Reply:
right the diff is bad here, it doesn't show changes below to where its being used because it was in a different function before.
happy with either solution.